bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/wrappers/classic/DataClientClassicApi.java

```diff
@@ -0,0 +1,238 @@
+        new BigtableTableName(
+            NameUtil.formatTableName(
+                requestContext.getProjectId(), requestContext.getInstanceId(), tableId));
+    return new BulkReadClassicApi(session.createBulkRead(tableName));
+  }
+
+  @Override
+  public ApiFuture<Void> mutateRowAsync(RowMutation rowMutation) {
+    ListenableFuture<MutateRowResponse> response =
+        delegate.mutateRowAsync(rowMutation.toProto(requestContext));
+    return ApiFutureUtil.transformAndAdapt(
+        response,
+        new Function<MutateRowResponse, Void>() {
+          @Nullable
+          @Override
+          public Void apply(@Nullable MutateRowResponse mutateRowResponse) {
+            return null;
+          }
+        });
+  }
+
+  @Override
+  public ApiFuture<Result> readModifyWriteRowAsync(ReadModifyWriteRow readModifyWriteRow) {
+    ListenableFuture<ReadModifyWriteRowResponse> response =
+        delegate.readModifyWriteRowAsync(readModifyWriteRow.toProto(requestContext));
+    return ApiFutureUtil.transformAndAdapt(
+        response,
+        new Function<ReadModifyWriteRowResponse, Result>() {
+          @Override
+          public Result apply(ReadModifyWriteRowResponse response) {
+            // TODO(rahulkql): should we create RowAdapter for proto Row to hbase's Result?
+            return Adapters.ROW_ADAPTER.adaptResponse(
+                new DefaultRowAdapter().createRowFromProto(response.getRow()));
+          }
+        });
+  }
+
+  @Override
+  public ApiFuture<Boolean> checkAndMutateRowAsync(ConditionalRowMutation conditionalRowMutation) {
+    final CheckAndMutateRowRequest request = conditionalRowMutation.toProto(requestContext);
+    final ListenableFuture<CheckAndMutateRowResponse> response =
+        delegate.checkAndMutateRowAsync(request);
+    return ApiFutureUtil.transformAndAdapt(
+        response,
+        new Function<CheckAndMutateRowResponse, Boolean>() {
+          @Override
+          public Boolean apply(CheckAndMutateRowResponse checkAndMutateRowResponse) {
+            return checkAndMutateRowResponse.getPredicateMatched();
+          }
+        });
+  }
+
+  @Override
+  public ApiFuture<List<KeyOffset>> sampleRowKeysAsync(String tableId) {
+    String fullTableName =
+        NameUtil.formatTableName(
+            requestContext.getProjectId(), requestContext.getInstanceId(), tableId);
+    SampleRowKeysRequest requestProto =
+        SampleRowKeysRequest.newBuilder().setTableName(fullTableName).build();
+    ListenableFuture<List<SampleRowKeysResponse>> responseProto =
+        delegate.sampleRowKeysAsync(requestProto);
+
+    return ApiFutureUtil.transformAndAdapt(
+        responseProto,
+        new Function<List<SampleRowKeysResponse>, List<KeyOffset>>() {
+          @Override
+          public List<KeyOffset> apply(List<SampleRowKeysResponse> rowKeysList) {
+            if (rowKeysList == null || rowKeysList.isEmpty()) {
+              return Collections.emptyList();
+            }
```

Comment:
I don't think rowKeysList can ever be null and if it's empty then we'll just return an empty ImmutableList below. I think its safe to remove it